cmd/steps/load_state.go

```diff
@@ -0,0 +1,91 @@
+package steps
+
+import(
+  "encoding/json"
+  "fmt"
+  "io/ioutil"
+  "log"
+)
+
+
+func loadState(command string) RunState {
+  filename := getRunResultFilename(command)
+  content, err := ioutil.ReadFile(filename)
+  if err != nil {
+    log.Fatal(err)
+  }
+  var serializedRunState SerializedRunState
+  err = json.Unmarshal(content, &serializedRunState)
+  if err != nil {
+    log.Fatal(err)
+  }
+  return RunState{
+    AbortStep: deserializeStep(serializedRunState.AbortStep),
+    RunStepList: deserializeStepList(serializedRunState.RunSteps),
+    UndoStepList: deserializeStepList(serializedRunState.UndoSteps),
+  }
+}
+
+
+func deserializeStep(serializedStep SerializedStep) Step {
```

Comment:
Let's try this separately after this is merged. I believe it was important to not have the type be `Step` for the reference of the object passed to `json.Unmarshal`. This is because `Step` is an interface and doesn't have the visible variables that the specific structs do.